### ISSUE TITLE
scan-data-for-missing - fails on cases

### ISSUE THREAD

**Issue by genmeblog** [techascent/tech.ml.dataset#84]
When first value is `nil` it's not recognized as missing

```clojure
(col/scan-data-for-missing ["1" nil])
;; => {:data ["1" ""], :missing [1]}

(col/scan-data-for-missing [nil "1"])
;; => {:data [nil "1"]}
```

**Comment by cnuernber** [comment]
Hmm, nil should always be missing regardless of the column type.  I wonder if this is related to your other issue about nil values.

**Comment by cnuernber** [comment]
That function (scan-data-for-missing) is pretty poor:

https://github.com/techascent/tech.ml.dataset/blob/d3b59477f192b100c0fe70713b90e0280ac4da8c/src/tech/ml/dataset/column.clj#L262

It doesn't use the parsing systems because those are setup for either spreadsheets for csv tables.  There is probably a way to gather the set of columns you like to scan and make a spreadsheet interface out of them but then it is just as efficient (and less work) to use the mapseq pathway (which is what the name-values-seq pathway does).  

Hmm, if that (use spreadsheet pathway) is really the best option it can't be that much work to build a fake sheet with a row with one cell.

**Comment by genmeblog** [comment]
I can go mapseq pathway, no problem. Just (again) fell into the trap of using `new-column` which require manual setup of datatype and missing values. I thought that `scan-data-for-missing` would give me that information out of the box...

**Comment by cnuernber** [comment]
It should, I am fixing this issue now.

**Comment by cnuernber** [comment]
Fixed in release 2.0-beta-51.
